Lib/cu2qu/rf.py

```diff
@@ -46,6 +44,103 @@ def zip(*args):
     return _zip(*args)
 
 
+class GetSegmentsPen(AbstractPen):
+    """Pen to collect segments into lists of points for conversion."""
+
+    def __init__(self):
+        self._last_pt = None
+        self.segments = []
+
+    def _add_segment(self, tag, *args):
+        self.segments.append((tag, args))
+
+    def moveTo(self, pt):
+        self._add_segment('move', pt)
+        self._last_pt = pt
+
+    def lineTo(self, pt):
+        self._add_segment('line', pt)
+        self._last_pt = pt
+
+    def curveTo(self, *points):
+        self._add_segment('curve', self._last_pt, *points)
+        self._last_pt = points[-1]
+
+    def closePath(self):
+        self._add_segment('close')
+
+    def addComponent(self, glyphName, transformation):
+        self._add_segment('component', glyphName, transformation)
+
+
+def _get_segments(glyph):
+    """Get a glyph's segments as extracted by GetSegmentsPen."""
+
+    pen = GetSegmentsPen()
+    glyph.draw(pen)
+    return pen.segments
+
+
+def _set_segments(glyph, segments):
+    """Draw segments as extracted by GetSegmentsPen back to a glyph."""
+
+    pen = glyph.getPen()
+    for tag, args in segments:
+        if tag == 'move':
+            pen.moveTo(*args)
+        elif tag == 'line':
+            pen.lineTo(*args)
+        elif tag == 'qcurve':
+            pen.qCurveTo(*args[1:])
+        elif tag == 'close':
+            pen.closePath()
+        elif tag == 'component':
+            pen.addComponent(*args)
+        else:
+            raise AssertionError('Unhandled segment type %s' % tag)
+
+
+def _segments_to_quadratic(segments, max_err, stats):
+    """Return quadratic approximations of cubic segments."""
+
+    assert all(s[0] == 'curve' for s in segments), 'Non-cubic given to convert'
+
+    new_points, _ = curves_to_quadratic([s[1] for s in segments], max_err)
+    n = len(new_points[0])
+    assert all(len(s) == n for s in new_points[1:]), 'Converted incompatibly'
+
+    n = str(n)
+    if stats is not None:
+        stats[n] = stats.get(n, 0) + 1
+
+    return [('qcurve', p) for p in new_points]
+
+
+def _fonts_to_quadratic(fonts, max_err, stats):
+    """Do the actual conversion of fonts, after arguments have been set up."""
+
+    for glyphs in zip(*fonts):
+        name = glyphs[0].name
+        assert all(g.name == name for g in glyphs), 'Incompatible fonts'
```

Comment:
This is fine for now, but later on we really want to support glyphs that are not present in all masters.  It's not hard to implement that here; just compute the union of all the glyph lists first, and process one at a time.